### ISSUE TITLE
FlatList doesn't call onEndReached if 'refreshing' is false

### ISSUE THREAD

**Issue by idibidiart** [facebook/react-native#14031]
### Description

If I set \the `refreshing` prop to false on FlatList it fails to call onEndReached. I tried different 0-1 values for the Threshold to no avail.

### Reproduction Steps and Sample Code

```
import React, { Component } from 'react';
import {
  FlatList,
  Text,
  ScrollView
} from 'react-native';

import { Row, Column as Col} from './grid'

import { MaterialIcons } from '@expo/vector-icons';

import faker from 'faker';

let j = 0
const randomUsers = (count = 10) => {
  const arr = [];
  for (let i = 0; i < count; i++) {
    arr.push({
      key: faker.random.uuid(),
      date: faker.date.weekday(),
      name: faker.name.firstName(),
      job: faker.name.jobTitle(),
      index: j++
    })
  }
  return arr
}

export default class Home extends Component {
  state = {
    refreshing: true,
    data: randomUsers(10),
  };

  onEndReached = () => {
    const data = [
        ...this.state.data,
        ...randomUsers(10),
      ]

    this.setState(state => ({
      data
    }));
  };

  onRefresh = () => {
    this.setState({
      data: randomUsers(10),
    });
  }

  render() {
    return (
      <ScrollView>
        <FlatList
          data={this.state.data}
          initialNumToRender={10}
          onEndReachedThreshold={1}
          onEndReached={this.onEndReached}
          refreshing={this.state.refreshing}
          onRefresh={this.onRefresh}
          renderItem={({ item }) => {
            return (
                <Row key={item.key} style={{paddingTop: '6%', paddingBottom: '6%', backgroundColor: 'white', borderBottomColor: 'lightgray', borderBottomWidth: 1}}>
                    <Col size={80} offset={6} >

                      <Row wrap>
                        <Col size={60} breakPoints={{sm: 200}}>
                          <Text style={{fontSize: 15, color: '#BD1206', fontWeight:'bold'}}>{String(item.date)}</Text>
                          <Row>
                            <Col size={10}>
                              <MaterialIcons name='person' size={17} color='gray'/>
                            </Col>
                            <Col smSize={60} size={87.5} offset={2.5}>
                              <Text style={{fontSize: 12, color: 'gray', lineHeight: 20}}>{item.job}</Text>
                            </Col>
                          </Row>
                        </Col>
                        <Col size={40} breakPoints={{sm: 200}}>
                          <Text style={{fontSize: 16, color: '#0a0a0a'}}>{item.name}</Text>
                        </Col> 
                     
                      </Row>    

                    </Col>
                    <Col size={14} offset={-6} hAlign='right'>
                          <Text>{item.index}</Text>
                    </Col>
                </Row>
            );
          }}
        />
      </ScrollView>
    );
  }
}
```

### Solution

If FlatList is built entirely in JS I might be able to help debug this issue. I'm also worried that my Row component might be doing something that interferes with FlatList. 

I've also noticed intermittent issue where FlatList is 0.75 of the way to the end of the currently loaded data set and it just gets stuck even if the threshold value is 0 or 1. But that's a different issue. Just pointing out that FlatList is probably Alpha quality at this stage, so it's more likely to be an issue in FlatList than my innocent looking code: https://github.com/idibidiart/react-native-responsive-grid/tree/master/src/components

### Additional Information

React Native 0.43
MacOS Seirra
iOS (latest, running in Simulator)
Expo SDK / XDE

**Comment by idibidiart** [comment]
@sahrens 

Summary:

FlatList fails to call onEndReached unless 'refreshing' is set to true. Tried different Threshold values from 0 to 1.

**Comment by sahrens** [comment]
You shouldn't wrap lists or other ScrollView insude ScrollViews - it just breaks things unless you are very careful with your styles.

If removing the wrapped ScrollView

**Comment by sahrens** [comment]
If removing the wrapping ScrollView doesn't fix the issue, feel free to update and reopen.

**Comment by idibidiart** [comment]
That did it @sahrens ... I had copied it from a source that used React Native Element's List component as a wrapper and so I just replaced with ScrollView thinking the scroll handling was done by the ScrollView and FlatLIst listened to events from ScrollView. Assumed FlatList was a pure JS implementation. Good to know! Thanks.

I'm going to run a few tests like giving items random height dynamically and seeing how scrollTo behavior copes with unpredictable item positions etc... and will keep kicking the tires! Looks good, Thank you.
